api/src/main/java/com/graphhopper/util/DistanceCalcEarth.java

```diff
@@ -149,6 +149,8 @@ public double calcNormalizedEdgeDistance3D(double r_lat_deg, double r_lon_deg, d
 
         double norm = delta_lon * delta_lon + delta_lat * delta_lat + delta_ele * delta_ele;
         double factor = ((r_lon - a_lon) * delta_lon + (r_lat - a_lat) * delta_lat + (r_ele - a_ele) * delta_ele) / norm;
+        if (factor < 0 || Double.isNaN(factor)) factor = 0;
```

Comment:
@boldtrn this fixed the Australia issue and fixed some unit tests I added, I saw that other code was not trimming segments (calcNormalizedEdgeDistance sets reduceToSegment to false) - is there a reason to retain that behavior here and only handle the NaN case?

Reply:
Thanks for finding this. I am not sure if we still need this. If I understand this right, the option was used to remain backward compatibility. WDYT @karussell?

Reply:
IMO we can remove the reduceToSegment parameter as it seems that only reduceToSegment==false is used. 

The option was introduced several years ago in this commit: https://github.com/graphhopper/graphhopper/commit/98e22f449f17a06db20296659b64011dbca9118e#diff-381813cce27dac5fde2afa44aec5bb13L164
but wasn't used it seems. Ugly.